src/main/host/process.rs

```diff
@@ -674,6 +679,79 @@ impl RunnableProcess {
     pub fn shmem(&self) -> impl Deref<Target = ShMemBlock<'static, ProcessShmem>> + '_ {
         &self.shim_shared_mem_block
     }
+
+    /// Emulate the `prlimit64` syscall for this process, operating on potentially both
+    /// native and emulated resource limits.
+    pub fn prlimit64(
+        &mut self,
+        resource: linux_api::resource::Resource,
+        new_rlim: Option<&linux_api::resource::rlimit64>,
+        old_rlim: Option<&mut linux_api::resource::rlimit64>,
+    ) -> Result<(), linux_api::errno::Errno> {
+        let idx = usize::try_from(u32::from(resource)).unwrap();
+        let cur = &self.common.rlimits[idx];
+        if let Some(old_rlim) = old_rlim {
+            *old_rlim = *cur;
+        }
+        let Some(new_rlim) = new_rlim else {
+            // Nothing else to do.
+            return Ok(());
+        };
+        if new_rlim.rlim_cur > new_rlim.rlim_max {
```

Comment:
nit: I think `new_rlim.rlim_cur > std::cmp::min(new_rlim.rlim_max, cur.rlim_max)` would be nicer. I think this is checked transitively through the `new_rlim.rlim_max > cur.rlim_max` below, but I think it would be nice to have this explicitly.